packages/langium/test/grammar/type-system/type-validator.test.ts

```diff
@@ -26,15 +26,20 @@ describe('validate params in types', () => {
         terminal ID: /[a-zA-Z_][\\w_]*/;
         `.trim();
         // verify we only have 1 error, associated with a missing 'name' prop
-        const document = await parseDocument(grammarServices, prog);
-        let diagnostics: Diagnostic[] = await grammarServices.validation.DocumentValidator.validateDocument(document);
-        diagnostics = diagnostics.filter(d => d.severity === DiagnosticSeverity.Error);
-        expect(diagnostics).toHaveLength(1);
-
-        // verify location of diagnostic
-        const d = diagnostics[0];
-        expect(d.range.start).toEqual({ character: 8, line: 4 });
-        expect(d.range.end).toEqual({ character: 10, line: 4 });
+        const validation = await validate(prog);
+
+        expectError(validation, "A property 'name' is expected. in a rule that returns type 'B'.", {
+            range: {
+                start: {
+                    line: 4,
+                    character: 8
+                },
+                end: {
+                    line: 4,
+                    character: 10
+                }
+            }
+        });
```

Comment:
This seems like an unnecessary change. I would roll this back, as it's not related to the PR. This change also alters the test semantics, making it looser than before (several diagnostics can occur, which was expressly forbidden before).